stdlib/email/src/main/java/org/ballerinalang/stdlib/email/util/EmailAccessUtil.java

```diff
@@ -0,0 +1,175 @@
+        properties.put(EmailConstants.PROPS_POP_AUTH, "true");
+        properties.put(EmailConstants.PROPS_POP_STARTTLS, "true");
+        properties.put(EmailConstants.PROPS_POP_SSL_ENABLE,
+                emailAccessConfig.getBooleanValue(EmailConstants.PROPS_SSL));
+        properties.put(EmailConstants.MAIL_STORE_PROTOCOL, EmailConstants.POP_PROTOCOL);
+        return properties;
+    }
+
+    /**
+     * Generates Properties object using the passed MapValue.
+     *
+     * @param emailAccessConfig MapValue with the configuration values
+     * @param host Host address of email server
+     * @return Properties Email server access properties
+     */
+    public static Properties getImapProperties(MapValue emailAccessConfig, String host) {
+        Properties properties = new Properties();
+        properties.put(EmailConstants.PROPS_IMAP_HOST, host);
+        properties.put(EmailConstants.PROPS_IMAP_PORT,
+                Long.toString(emailAccessConfig.getIntValue(EmailConstants.PROPS_PORT)));
+        properties.put(EmailConstants.PROPS_IMAP_AUTH, "true");
+        properties.put(EmailConstants.PROPS_IMAP_STARTTLS, "true");
+        properties.put(EmailConstants.PROPS_IMAP_SSL_ENABLE,
+                emailAccessConfig.getBooleanValue(EmailConstants.PROPS_SSL));
+        properties.put(EmailConstants.MAIL_STORE_PROTOCOL, EmailConstants.IMAP_PROTOCOL);
+        return properties;
+    }
+
+    /**
+     * Generates MapValue object using the passed message.
+     *
+     * @param message Email message received
+     * @return MapValue Ballerina compatible map object
+     * @throws MessagingException If an error occurs related to messaging
+     * @throws IOException If an error occurs related to I/O
+     */
+    public static MapValue getMapValue(Message message) throws MessagingException, IOException {
+        Map<String, Object> valueMap = new HashMap<>();
+        BArray toAddressArrayValue = getAddressBArrayList(message.getRecipients(Message.RecipientType.TO));
+        BArray ccAddressArrayValue = getAddressBArrayList(message.getRecipients(Message.RecipientType.CC));
+        BArray bccAddressArrayValue = getAddressBArrayList(message.getRecipients(Message.RecipientType.BCC));
+        BArray replyToAddressArrayValue = getAddressBArrayList(message.getReplyTo());
+        String subject = getStringNullChecked(message.getSubject());
+        String messageBody = "";
+        if (message.isMimeType("text/plain")) {
+            if (message.getContent() != null) {
+                messageBody = message.getContent().toString();
+            }
+        } else if (message.isMimeType("multipart/*")) {
+            MimeMultipart mimeMultipart = (MimeMultipart) message.getContent();
+            messageBody = getTextFromMimeMultipart(mimeMultipart);
+        }
+        String fromAddress = "";
+        if (message.getFrom() != null) {
+            fromAddress = message.getFrom()[0].toString();
+        }
```

Comment:
IMO, we can move these to another smaller methods as well.

Reply:
Broken into small methods.